### PR TITLE
Ports Delta Station

### PR DESCRIPTION
**Pull request by SamHPurp**
So this has been painstakingly added tile by tile, item by item from scratch starting with an empty file - because TG's systems are so different to us, this seemed at the time easier to do. (It probably wasn't).

--------

# **FAQ:**

**What IS Delta Station on Paradise:**

A larger station with more room to play around in. It is old, rusted and dirty. It has an area for a Supermatter shard for atmospherics techs to ~~blow everyone up with~~ play around with. It has a combined Bar/kitchen area, which also has a stage for people to put on shows ~~meatspike clowns~~ and all sorts of other misc.

It has *currently* a couple of areas that are useful for **if** Lavaland gets implemented. There are nice abandoned areas through the station that enterprising young greytide can do up and make look swanky.

The brig is going to need to be re-done, but at the moment, all departments have security checkpoints.

Each job (and a few bonus ones) have toy figures in their office. Great for collectors. ;)

**What Delta Station on Paradise ISN'T:**

Be a verbatim port. It has had and will require further tweaks and overhauls of areas.

**Why have you modified Box's Z2?**

This was necessary to expand the landing zone of the shuttle both at Centcomm and when Hijacked.

**This is awesome, how can I help?**

Report **bugs** in the comments below and pester Headmins & Maintainers for test-merges.

-------

Props to @owenowen212 for the following image:
image

--------

And of course credit for the original design (which is likely to be heavily modified for Paradise): @Okand37

Closes: #8274

---------
:cl: Purpose
add: Delta Station, codenamed Kerberos has been acquired by Nanotrasen.
/:cl:

### CODE REVIEW

**Comment by Kyep** [comment]
For the benefit of those of us not familiar with delta... could you summarize the main advantages this map has over boxstation?
Besides "supports more players".

**Comment by Okand37** [comment]
Neat. I've never played on paradise's codebase but I noticed you've blocked off the auxiliary mining base-or atleast, the area that is intended to be that. Might I suggest just replacing it with a standard free construction area opposed to just barricading off the space? Could be fun!

I don't know whether to be concerned for your health or applaud you for your patience in recreating it, but good job either way! Hope it was fun!

**Comment by SamHPurp** [comment]
@Okand37 I'm more concerned about how you did it in the first place, you poor poor man.

We're working on porting Lavaland, so Aux mining base will become used, eventually. But yeah in the meantime it'll probably just be a construction area for people to do bits and bobs in.

**Comment by taukausanake** [comment]
@SamHPurp This is amazing! I can't believe you managed to do all this and not die. I never thought we'd be porting other maps in. I've been messing with simple maps like Ministation myself just because. I so want to try playing on this map some time

**Comment by SamHPurp** [comment]
So, theres still a significant amount of work left to do, which I'll try and break down my current thoughts on this:

We lack the Paradise karma roles on here currently. Barber, pod pilot, blueshield, nt rep, mechanic etc... also brig is too small for Paradise's needs... and the sec checkpoints also probably aren't right for Paradise.

It has on-station Telecomms... so a solution may be that I tear out bits of the brig, re-do it bigger and better... remove sec-checkpoints... move telecomms back to the satellite... allowing us to then add in all these things.

Also Medbay has only 1 OR right now, we gotta fix that.

**Comment by taukausanake** [comment]
I say keep the security checkpoints. I don't want maps to just be remaps of each other, I want different functionality. Different us-abilities. Same for Telecomms. Then you can have the old Telecomms satellite as a ruins that antags can use. That's what the other server stations have sometimes. Even have backstory to it

**Comment by SamHPurp** [comment]
@taukausanake locating all the Paradise karma bits without those sorts of modifications is going be unreal that way... hm

**Comment by taukausanake** [comment]
This is a map designed with a whole lot of clutter design-wise. If you're unwilling to remove those parts at all then that's one thing but if not before the checkpoints then that's another. Besides, removing the checkpoints isn't going to make a whole ton of space for much; can't even justify moving everything else into the empty space to make room in other areas.

Mind if I try looking at places and submit feedback? Can I post images in the Redpen.io? I'd like to help out if at all

**Comment by SamHPurp** [comment]
Yup, all makes sense.

Images and stuff may be more usefully posted over on the Trello: https://trello.com/b/SNjxuPvn/delta-station ... you might need an invite to be able to post on it tho... if so, just msg me on Discord.

**Comment by Aurorablade** [comment]
i don't really like on station tcomms...So yeah some things need to be cut for karma roles.

**Comment by scrubmcnoob** [comment]
Looking great. Hope to see this merged in at some point.
Thanks for the effort into this project.

**Comment by SamHPurp** [comment]
@Aurorablade I agree with you on that.

@scrubmcnoob Yup, thats on the cards!

**Comment by Citinited** [comment]
This looks great, just leaving a couple remarks on your pinboard.

Also: roundstart supermatter, nice to see you again!

**Comment by KingPhilipIII** [comment]
I like the idea of checkpoints as well. They aren't really utilized on TG all that much as far as the brig cells, but for us, who are a bit more meticulous in arresting it could be nice to just brig some unruly punk for five minutes on site instead of dragging them all the way back, and bringing more serious issues back to the brig to be processed.

I play on TG a lot myself and I personally love how much maintanence it has. Lots of places to duck and hide.

**Comment by taukausanake** [comment]
Okay so I'm too damn lazy to make a new account on a thing to make posts when GitHub is just fine so I'm going to post my suggestions here.

For the paramedic I think the only place you can put it is here below Chemistry, West of Cloning.
image

There seems to be a missing Mr. Changs? I can't find it at least. Maybe put that in one of the forgotten rooms in maint.

Barber could maybe go in this place right below the Library. Add a wall above the hallway maint door and add in another maint door so you can access the tunnels from the Library
image
Else you might be able to squeeze it in here below the crew quarters showers but that might be too tough.
image

There is honestly no place in Engineering you can put the Mechanic, not where they won't have to walk through secure areas in Engineering like any of the engines there. My suggestion: put the Mechanic in the area East of Arrivals, that big open area. Maybe it can also be a good sized garage too, more than enough space for the Mechanic to build pods. Even better it's already behind Engineering access.
image
Or you can make the area below it the Mechanic bay and the empty area t pod garage like Mechbay is to Robotics. Might as well too seeing as there isn't really a good spot in Escape for a pod bay, not without moving the Security checkpoint around.

IAA doesn't have an office it looks. Might be able to go somewhere near the Courtroom but can't make a good call on that. Maybe make a space in the Command section of the station, looks like of spacious in there.
Sounds like Security will be going over some work though so maybe it'll sort itself out. Same for the Brig Physician and Sec Pod Pilot. That's like half the karma roles there.


I think that's all from me for now

**Comment by owenowen212** [comment]
Hooray, hooray Delta is on the way 👌.  Major props for the effort you've put into this so far @SamHPurp . 👍

On the off chance that this may be useful for people in identifying what particular rooms are, I'll leave this below.

**Comment by SamHPurp** [comment]
Bugger I had planned on including that in the OP (Crediting you of course) and I entirely forgot!

**Comment by taukausanake** [comment]
image
Awww look it's happy ❤️

**Comment by SamHPurp** [comment]
First batch of Paradising. Adds three of the karma roles.... more to come, just need to temp shift attention to Meta.

**Comment by ModDownloading** [comment]
Oho! You lads are adding my favorite station! I sincerely hope this goes through!

(Even if the reason WHY this is my favorite station is because it crams so much stuff into one mass of metal)

**Comment by SamHPurp** [comment]
Paradise's maintainers do not wish to maintain another map. As a result, this is being closed.

I'll leave my branch available for anyone that is able to convince them otherwise.

**Comment by SamHPurp** [comment]
After a number of ~~spoiled self-entitled brats~~ people complained about this closing, I've gone and updated it.

Adding Mechanic and Pod Pilot... to their own hanger area. A different design than we've seen before, but should be quite neat.

Pic coming.

**Comment by FreeStylaLT** [comment]
🙌 bah gawd

**Comment by Ralta** [comment]
Yeeeeeeeeeeeeeeeees. 🙏

**Comment by SamHPurp** [comment]
@TullyBurnalot @tigercat2000 Do my a favour and let me know before the first playtest or so.

**Comment by taukausanake** [comment]
I thought this was abandoned because map maintenance? Glad to see it back though.

I totally wasn't one of those people who complained. I just forked it to my own branch. :^) After fixing violin and poster paths

**Comment by ParemberKennard** [comment]
Found a problem with drones. The only fabricator is on the Engineering Outpost, and the shuttle connection between outpost and station doesn't work. So as is, if you spawn as a drone, you're stuck on the engineering shuttle without admin help.

**Comment by KingPhilipIII** [comment]
Right. Myriad of map issues that were originally in the issues section but Fox shooed us away so I'm here to report them. Some I found myself through my own investigations, others were discovered by others.

1. No cryodorms mapped in. The problem here should be apparent.

2. The vault has no access requirements. Anyone can walk in and steal stuff if it's not bolted.

3. The janitor's shutter button doesn't have access requirements, once again allowing unfettered theft(Filthy greytide now has cleaning grenades.)

4. The kitchen has neither shutters nor glass windoors to keep unruly crew(Redirects to Greytide) away from the food and more importantly sharp knives.

5. Some of the chapel's doors at roundstart are seethrough, as though they are glass airlocks, despite them not being so. Opening and closing them seems to fix this however.

6. The brig cells in the brig do not have timers, making prisoner management a pain in the ass.

7. We're missing a whole bunch of wardrobe stuff in dorms for all of our Snowflakey crew to dress up.

8. Delta has two AI uploads. The original TG map design had telecommunications inside the bridge, placing the upload as part of the satellite. We have the telecomms satellite, so filling the space in Paranerds added in an AI upload.

9. The barbershop has gone from unused and unloved to unused, unloved and unlit. There are no lightbulbs or light tubes in the barber shop.

**Comment by SamHPurp** [comment]
@KingPhilipIII cheers

**Comment by taukausanake** [comment]
For number 8, could we have telecomms on the station where it should be but also have the abandoned telecomms satellite on its z-level? I think that would be best, give variation to the maps

**Comment by KingPhilipIII** [comment]
Oh. Quick #9. The door to the bridge AI upload says messaging server, as that used to lead to the telecomms control room.

**Comment by SamHPurp** [comment]
@taukausanake to mimick Meta?

**Comment by taukausanake** [comment]
@SamHPurp Huh. I guess we already have that setup? But even then, what are you going to do with either the AI satellite or the upload in the bridge?

@KingPhilipIII PR 9 - Shotgun and tweaks?

**Comment by KingPhilipIII** [comment]
@taukausanake I just meant that as an addition to my list. didn't mean to reference an ancient PR.

**Comment by taukausanake** [comment]
I know, I just get a little triggered when it happens

**Comment by KingPhilipIII** [comment]
@taukausanake Fixed.

**Comment by taukausanake** [comment]
@KingPhilipIII It doesn't matter, the damage was done. The PR will forever be reference. :frowning: 
That's why it triggers me so

**Comment by Spartan6** [comment]
Since I was apparently supposed to report all of these things in an issue but maintainers did not like that one bit, I’m confused and such, but here’s what I posted.

Problem Description:

- No timers on cells.
- There is laughably so few cells.
- The armoury is wide open and laid out terribly.
- There are two brigbays.
- There is a massive security office that serves no purpose.
- There is no processing.
- There is a massive meeting room that serves no purpose.
- CAMERAS DO NOT WORK AAAAA: The Camera console displays zip cameras.

What did you expect to happen:
Brig to function.

What happened instead:
*scream

Why is this bad/What are the consequences:
*SCREAM

Steps to reproduce the problem:
DELTA STATION

Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?): *DEATHGASP

**Comment by SamHPurp** [comment]
> No cryodorms mapped in. The problem here should be apparent.
Fixed
> The vault has no access requirements. Anyone can walk in and steal stuff if it's not bolted.
Fixed
> The janitor's shutter button doesn't have access requirements, once again allowing unfettered theft(Filthy greytide now has cleaning grenades.)
Fixed
> The kitchen has neither shutters nor glass windoors to keep unruly crew(Redirects to Greytide) away from the food and more importantly sharp knives.
Fixed
> Some of the chapel's doors at roundstart are seethrough, as though they are glass airlocks, despite them not being so. Opening and closing them seems to fix this however.
Fixed
> The brig cells in the brig do not have timers
Fixed
> missing wardrobe stuff in dorms for all of our Snowflakey crew to dress up.
~~Feature~~ Fixed
> The barbershop unlit and unpowered
Fixed
>Pod pilot is locked in, with a wall in his doorway
Fixed
>Pod pilot has the wrong flooring in one of his doors
Fixed
>duplicate APC in the captain's office
Fixed by marking his bedroom as a separate area
>The armoury is wide open and laid out terribly.
Fixed, and altered layout a bit.
>CAMERAS DO NOT WORK AAAAA: The Camera console displays zip cameras.
Fixed
>no drone fabricator on station
Fixed
>The door to the bridge AI upload says messaging server
Fixed
>No messaging server
Fixed

**Comment by SamHPurp** [comment]
@Allfd @TullyBurnalot @tigercat2000 This is ready for another play test, all reported bugs fixed.

**Comment by SamHPurp** [comment]
Reminder to self: Remove edits to Meta z3

**Comment by SamHPurp** [comment]
There is a bug with carpet smoothing and piping underneath. That'll need to be fixed as a separate thing, but I resolved any instances of that on the station.

Runtimes dog bed is now a kitty basket.

Botanists and Chefs are now allowed to leave their workplaces.

**Comment by KingPhilipIII** [comment]
Sammy, EVA doesn't have the entertainer spess suits. Call it a feature, call it a mapping issue, but it doesn't.

**Comment by Mitchs98** [comment]
Problem Description:
Several things are missing, mostly from maint. And the holodeck doesn't work at all.

Starters, rubber duckies are no where to be found on the station at all that I've seen. Like, not a single one. Not in maints, no where.

No secondary booze-o-mat. Now literally impossible to set up your own secondary bar without cargo.

Metal. Cargo lacks it at round start.

Pretty sure Kitchen lacks an oven?

We're missing SEVERAL of the clothes vendors.

Cryodorms.

What did you expect to happen:
For this stuff to be there.

What happened instead:
It was not.

Why is this bad/What are the consequences:
Mostly not really round breaking or anything. I just like the stuff.

Steps to reproduce the problem:
Load the map.

Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?): idk

**Comment by Ty-Omaha** [comment]
https://image.prntscr.com/image/mH5CFH2QQRy8rEwj6EJbYg.png
There's some space that isn't "moving" when the shuttle is in transit below the cockpit

**Comment by Squirgenheimer** [comment]
(I did was not able to play on the most recent playtest so I am not sure if this was fixed)

1. Many camera names in the AI's menu, or in chat when alarms go off, are mismatched. Example; 
https://image.prntscr.com/image/ViYHq4WNT6_He5irFxaPlA.png

2. The AI upload below the bridge, and every room except the AI core in the AI satellite did not appear to have turret controls.

**Comment by TDSSS** [comment]
Why is maint's floor this patchwork of floortiles?

**Comment by SamHPurp** [comment]
Added second booze-o-mat - to the gambling den.
Added three rubber duckies.
Added oven to the kitchen (what, you guys don't want 2 lots of deep fat fried food, I thought you were American?)
Added some Metal to Cargo
Added rusted walls and r-walls.
Added more dirt and fungus.
Fixed the Metal & Glass that spawn in Robotics.
Fixed holodeck
Fixed floor inconsistencies in Rec area
Fixed floor inconsistencies in Cryodorms
Fixed door access issue in Botany/Chef
Remapped swathes of atmos piping due to the previously mentioned carpet smoothing issue
Tweaked number of maintenance lights
Added sleeper circuitboard to tech storage
Added box of prisoner ID's to Security
Added sleeper to Brigbay

**Comment by KingPhilipIII** [comment]
So Sammy. Did you fix the double uploads yet?

**Comment by SamHPurp** [comment]
@KingPhilipIII Not 100% on the best resolution for it yet.

Could stick T-Comms on the one below AI Sat, but that'd have wider reaching consequences

**Comment by SamHPurp** [comment]
So as I've got zero bug reports from last nights playtest, I can assume that Tully broke it and the playtest didn't happen? 😆

**Comment by KingPhilipIII** [comment]
Sammy, we’re still missing blast doors I think from a good bit of the station.

**Comment by SamHPurp** [comment]
@KingPhilipIII examples?

**Comment by KingPhilipIII** [comment]
Unless you fixed it and I missed the note, all of them. The only ones that are actually important outside of a general lockdown though are the brig’s prison blast doors and the bridge.

**Comment by SamHPurp** [comment]
@KingPhilipIII I'll have a quick review of them, but some are intentionally missing ;)

**Comment by KingPhilipIII** [comment]
I mean, feature, bug, call it what you will just thought it was worth mentioning.

**Comment by Denthamos** [comment]
The plastic flaps in cargo to the cargo shuttle need to be replaced with the airtight versions, something I noticed today as the ones currently there are the normal ones.

**Comment by SamHPurp** [comment]
Lets see how many million conflicts this gets

**Comment by KasparoVy** [comment]
Couple million, not too shabby

**Comment by SamHPurp** [comment]
Jealous @KasparoVy ?

**Comment by Allfd** [comment]
You are a hero

**Comment by dapocalypse** [comment]
We should fix up Metastation before merging delta.

**Comment by Spartan6** [comment]
@dapocalypse Both can be done.

**Comment by variableundefined** [comment]
This PR has been approved for testing. This took long enough anyway.

The only way to test it is to have it merged in and tested. 

So that's what we will do.

**Comment by Ty-Omaha** [comment]
😮

**Comment by KasparoVy** [comment]
Holy shit nice

**Comment by Allfd** [comment]
FAKE NEWS!

**Comment by Spartan6** [comment]
I think there’s a blood moon eclipse

**Comment by variableundefined** [comment]
Canvas is causing everyone's Travis to fail, putting up a fix

**Comment by SamHPurp** [comment]
WHAT THE FUCK

WHAT TEMPERATURE IS HELL RIGHT NOW?

AS ANYONE SEEN FLYING PIGS?

**Comment by Calecute** [comment]
I'll give you my feedback from the playtest, mainly from atmos:

- Atmos is pretty big, I liked all the space and the supermatter. 
- Really like the action figure
- I actually liked that the filters are not set up at all
- The two atmos central computers are a god send, especially considering that with a bigger atmos room and a new big SM room you can be working atmos farther away from atmos control room than before
- I liked that we now have a heater set up for the air supply.
- I feel we have fewer customization options regarding the waste thou. Space around atmos is really cramped and I don't really see a convenient way to get waste through a space loop with the way it's piped now. At the same time we don’t have lots of coolers in a place we can throw waste in like box. Maybe we'll find a way to work around it thou 
- The Mix Tank Control is not set up properly, I think it should be.
- I'd really like the option to bypass the ports when sending mix to turbine. 
- The places where atmos connects to station air supply and scrubber networks are pretty unintuitive, I'm not completely sure what the constrains are here, but I think it could probably be done better.
- It's really unfortunate that the place you're supposed to set up the emitter can fire through the whole SM room into atmos proper
-Also it's unfortunate that you can laser the SM from a random corridor in maint.
- Didn't pay attention to the way SM wiring is set, but emitter has no wire to conect it and atmos can't touch hot wires. So if there's no way to turn off emitter power through the SMES we need the wire to conect the emitter already there. 
- Atmos external access was bolted and had no controls, and also atmos did not have access to it.
- Atmos should have EVA access to jetpacks.
- We need more hardsuits. Even box station has 4 hardsuits now, I imagine with the bigger map we have more atmos slots.
- Atmos needs multitool. It's required for a lot of more complex atmos things, like fixing the not properly set up mix tank control, or making some more complex stuff with AAC, pretty sure people will start using AAC with the SM.
- I really miss the welding googles, and we ran out of eye protection pretty quickly. We need at least enough eye protection for all the slots, disposals fixing is atmos job and requires it. 
- I really miss our alarms camera. It's really useful to take a look at a place and see what the problem is. For example, mining asteroid has an air alarm, and pressure is dropping. Was it just an airlock misuse, or did a Ripley drilled through the doors? If its airlock misuse you can just turn atmos back on and it will be alright, but if it’s breached someone will need to go fix it. You can wait to see if pressure gets stable or keep dropping, but atmos takes some time to work through, especially if it’s a big room. Or you can call a suspected breach and get someone to check. So you either waste time waiting before responding, or waste resources responding to things that didn't need boots on the ground.
- There's a solar that as far as I can tell can only be access through atmos. This will be a little inconvenient, as engies have no access to atmos, and atmos have no acess to the solar itself. Looking at the map there really doesn’t seem to be a way around it, and I don’t think we should change the access because of this, but if you see a way to fix it so that engies can go into the solar without going through atmos it would be nice
- Air alarm map seemed strange to me. Lots of rooms didn't have an air alarm on my map. 
- Some trash was ending up at atmos control room. Not sure if disposals is not set up properly or if someone fucked with it. 
- The disposal room itself was a complete and utter mess, nothing was working properly and there was trash everywhere. Again, not sure if that's a design problem or sabotage.

**Comment by SamHPurp** [comment]
@Calecute Thank you for your feedback on this, some valuable things in there I need to get on and fix!

**Comment by Calecute** [comment]
I played again on Delta, this time as civ and really liked the maint. I also ended up noticing a problem. I found two disposals sorters that had different name and tag. The sorter to library is called library but tagged chapel. One of the sorters going to science is named RD office and tagged robotics, or the opposite, can’t really remember. Also don't know where it ends up. 

Also that round breaches caused the loss of all air in atmos gas chambers according to the atmos techies. I really like that, always felt air should be more precious.

And something else I forgot to let you know from my atmos round. Apparently the air alarm in maint next to chapel is called "arrivals north maintenance air alarm" for some reason. 

Thanks for all the work mate.
